common/lib/dependabot/dependency_group.rb

```diff
@@ -22,15 +22,21 @@ class DependencyGroup
     sig { returns(T::Array[Dependabot::Dependency]) }
     attr_reader :dependencies
 
+    sig { returns(String) }
+    attr_reader :applies_to
+
     sig do
       params(
         name: String,
+        applies_to: T.nilable(String),
         rules: T::Hash[String, T.untyped]
       )
         .void
     end
-    def initialize(name:, rules:)
+    def initialize(name:, applies_to:, rules:)
       @name = name
+      # For backwards compatibility, if no applies_to is provided, default to "version-updates"
+      @applies_to = T.let(applies_to || "version-updates", String)
```

Comment:
👍

Reply:
Consider using as a default parameter in the function signature itself?

Reply:
I do need to set a default, but I also have to have this here because in Ruby the default parameter only applies when the value isn't set, so a `nil` value will still get through:

```ruby
irb(main):001* def f(v: "default")
irb(main):002*   v
irb(main):003> end
=> :f
irb(main):004> f()
=> "default"
irb(main):005> f(v: nil)
=> nil
irb(main):006> f(v: "hello")
=> "hello"
```

Reply:
minor: could Sorbet help with this?

Reply:
I think Sorbet is doing all it can here, there just are scenario's where we will pass `nil`, for example in GHES where the API has not been updated, we will send job definitions without an `applies-to` value and this is the entry point where we normalize the data to make it backwards compatible